source/unix/openssl_aes.c

```diff
@@ -0,0 +1,741 @@
+/**
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0.
+ */
+#include <aws/cal/symmetric_cipher.h>
+
+#include <openssl/evp.h>
+
+struct openssl_aes_cipher {
+    struct aws_symmetric_cipher cipher_base;
+    EVP_CIPHER_CTX *encryptor_ctx;
+    EVP_CIPHER_CTX *decryptor_ctx;
+    struct aws_byte_buf working_buffer;
+};
+
+static int s_encrypt(
+    struct aws_symmetric_cipher *cipher,
+    const struct aws_byte_cursor *input,
+    struct aws_byte_buf *out) {
+    size_t required_buffer_space = input->len + cipher->block_size;
+    size_t available_write_space = out->capacity - out->len;
+
+    if (available_write_space < required_buffer_space) {
+        if (aws_byte_buf_reserve_relative(out, required_buffer_space) != AWS_OP_SUCCESS) {
+            return aws_raise_error(AWS_ERROR_SHORT_BUFFER);
+        }
+    }
+
+    available_write_space = out->capacity - out->len;
+    struct openssl_aes_cipher *openssl_cipher = cipher->impl;
+
+    int len_written = (int)(available_write_space);
```

Comment:
```suggestion
    int len_written = 0;
```

it's weird to initialize something to a specific number when it's going to get overwritten. It also confuses me why you picked `required_buffer_space` over `available_write_space`, and also this is a naive cast to int.

so simplest to just set it 0

Reply:
it's more of an input/output size. For some versions of openssl, this value needs to be the original buffer capacity and then they overwrite it with the amount they wrote. However, i did just refactor all of this so....